fe1-web/__tests__/model/network/FromJson.ts

```diff
@@ -0,0 +1,496 @@
+import { sign } from "tweetnacl";
+import { encodeBase64 } from "tweetnacl-util";
+
+const assert = require('assert');
+const assertChai = require('chai').assert;
+
+const STALE_TIMESTAMP = new Timestamp(1514761200);    // 1st january 2018
+const STANDARD_TIMESTAMP = new Timestamp(1609455600); // 1st january 2021
+const CLOSE_TIMESTAMP = new Timestamp(1609542000);    // 2nd january 2021
+const FUTURE_TIMESTAMP = new Timestamp(1735686000);   // 1st january 2025
+
+// FIXME use real keys from key pair once operational
+export const mockPublicKey = new PublicKey('xjHAz+d0udy1XfHp5qugskWJVEGZETN/8DV3+ccOFSs=');
+export const mockSecretKey = new PrivateKey('vx0b2hbxwPBQzfPu9NdlCcYmuFjhUFuIUDx6doHRCM7GMcDP53S53LVd8enmq6CyRYlUQZkRM3/wNXf5xw4VKw==');
+
+
+const _generateKeyPair = () => {
+  const pair = sign.keyPair();
+  const keys = { pubKey: encodeBase64(pair.publicKey), secKey: encodeBase64(pair.secretKey) };
+  return { pubKey: new PublicKey(keys.pubKey), secKey: new PrivateKey(keys.secKey) };
+};
+
+
+
+
+
+describe('=== fromJson object checks ===', function() {
+
+  const sampleKey1: PublicKey = _generateKeyPair().pubKey;
+  const sampleKey2: PublicKey = _generateKeyPair().pubKey;
+
+  const mockMessageId = Base64Data.encode('message_id');
+
+  const org = mockPublicKey;
+  const time = STANDARD_TIMESTAMP;
+  const name = 'PoP\'s team "LAO" or filling name';
+  const mockLaoId: Hash = Hash.fromStringArray(org.toString(), time.toString(), name);
+
+  let temp: any = {};
+
+  const sampleCreateLao: Partial<CreateLao> = {
+    object: ObjectType.LAO,
+    action: ActionType.CREATE,
+    id: mockLaoId,
+    name: name,
+    creation: time,
+    organizer: org,
+    witnesses: [],
+  };
+
+  const sampleUpdateLao: Partial<UpdateLao> = {
+    object: ObjectType.LAO,
+    action: ActionType.UPDATE_PROPERTIES,
+    id: Hash.fromStringArray(org.toString(), time.toString(), name),
+    name: name,
+    last_modified: CLOSE_TIMESTAMP,
+    witnesses: [sampleKey1, sampleKey2],
+  };
+
+  const sampleStateLao: Partial<StateLao> = {
+    object: ObjectType.LAO,
+    action: ActionType.STATE,
+    id: Hash.fromStringArray(org.toString(), time.toString(), name),
+    name: name,
+    creation: time,
+    last_modified: CLOSE_TIMESTAMP,
+    organizer: mockPublicKey,
+    witnesses: [sampleKey1, sampleKey2],
+    modification_id: Hash.fromStringArray(mockMessageId.toString()),
+    modification_signatures: [],
+  };
+
+  const sampleCreateMeeting: Partial<CreateMeeting> = {
+    object: ObjectType.MEETING,
+    action: ActionType.CREATE,
+    id: Hash.fromStringArray(mockLaoId.toString(), time.toString(), name),
+    name: name,
+    creation: time,
+    location: 'Lausanne',
+    start: time,
+    end: FUTURE_TIMESTAMP,
+    extra: { extra: 'extra info' },
+  };
+
+  const sampleStateMeeting: Partial<StateMeeting> = {
+    object: ObjectType.MEETING,
+    action: ActionType.STATE,
+    id: Hash.fromStringArray(mockLaoId.toString(), time.toString(), name),
```

Comment:
technically, if we're looking at a JSON, id is a string, not a `Hash` object. You might want to use the `.valueOf()` method.
Then of course, we're stuck with untyped objects, which is the (sad) reality of untrusted data from the user, servers, etc.